### ISSUE TITLE
Empty all val graphs

### ISSUE THREAD

**Issue by Jes46** [ultralytics/ultralytics#7345]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

Hello, I've made a training with YOLOv8n 3 times and all of them I got the 3 **val graphs** empty like you can see
![results](https://github.com/ultralytics/ultralytics/assets/150959706/1c349e53-789d-4c7b-a7fc-057919e48c59)
And after the training I always see the next line with **loss value** always to 0
![loss](https://github.com/ultralytics/ultralytics/assets/150959706/e86f77da-0446-415a-a5cd-f76fe490e480)

The case is my model is working good and I've used the validation function as well and I got all de curves (F1, P, PR and R) with information so I don't know what's going on. 
![R_curve](https://github.com/ultralytics/ultralytics/assets/150959706/2e073900-f2ce-4244-8aa9-f29f51610a5a)
![F1_curve](https://github.com/ultralytics/ultralytics/assets/150959706/68239174-c605-4348-a89f-cd891c6f7927)
![P_curve](https://github.com/ultralytics/ultralytics/assets/150959706/7f1a4b70-462d-43b7-b17c-8907dd88fec6)
![PR_curve](https://github.com/ultralytics/ultralytics/assets/150959706/907227de-fcf3-400e-af51-d552b97dbb58)

Besides, like you can see here these are the results in the .csv file where you can see "nan" value, maybe here is the issue.
![nan boxs](https://github.com/ultralytics/ultralytics/assets/150959706/fdb438b9-9a76-4f6d-8459-b65eedcf2916)

### Environment

Visual Studio Code
Python 3.9.12 64 bits

### Additional

Here is my database.yaml, I'm using my own database and the code where I make the call to training.
**Dataset.yaml**
![yaml](https://github.com/ultralytics/ultralytics/assets/150959706/f394a378-f632-439c-b08b-59f4f547b8db)
**Train code**
![train](https://github.com/ultralytics/ultralytics/assets/150959706/862be293-6acf-4f76-9c23-64c7403b98bf)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@Jes46 hello! Thanks for providing detailed information about your issue with the validation graphs during YOLOv8 training. It's great to hear that your model is performing well despite the empty val graphs.

The presence of "nan" values in your .csv file suggests there might be an issue with the validation data or the way metrics are being logged. This could be due to a variety of reasons, such as incorrect data labeling, data preprocessing issues, or a bug in the logging process.

Since your other metrics like F1, Precision, Recall, and PR curves are showing information, it's likely that the problem is isolated to the specific part of the code responsible for generating the val graphs.

To troubleshoot this, please ensure that your dataset is correctly formatted and that all labels are correct. Also, verify that your `dataset.yaml` is properly configured with the correct paths and settings for your custom dataset.

If the issue persists, consider reviewing the validation code to ensure it's processing the data correctly and that the logging mechanism is functioning as expected.

If you need further assistance, please provide more details about your dataset and the exact steps you're following in the training process. You can refer to our documentation for additional guidance on training and validation.

We appreciate your contribution to the YOLOv8 community, and we're here to help you get to the bottom of this issue! 🚀

**Comment by Jes46** [comment]
Thanks for your answer. About my **_dataset.yaml_** these are my directories where I have all my dataset.
![Directorio Base de Datos](https://github.com/ultralytics/ultralytics/assets/150959706/51a26a5e-ef23-489b-b4d1-052f0b30e425)
![yaml](https://github.com/ultralytics/ultralytics/assets/150959706/f3f1784e-c8de-4e2a-bd46-c7e67c8d5aa5)

**Important! ->** Both folders contain the same images and the same label files. I thought it would be a fast option to make my firts trainings and test configurations. Maybe It's the main problem.

**Important! ->** My training code line is the following: 
model.train(data=data_path, epochs=100, batch=4, project='Other', name='Jesus', imgsz=640, optimizer='Adam', **_amp=False_**)
Is there any possibility amp=False was the problem?

Then, when you said "consider reviewing the validation cod" -> How do I do that? I'm using the .val() function and the graphs are generated in training mode.

And finally, also where you say "t's likely that the problem is isolated to the specific part of the code responsible for generating the val graphs." -> Are there any hyperparameter that make special influence on that loss graphs?

Thanks for your help.

**Comment by glenn-jocher** [comment]
@Jes46, it's good to have more details about your setup. Using the same images and labels for both training and validation is not recommended as it can lead to overfitting and might not provide a true measure of your model's performance on unseen data. It's best to have a separate, diverse validation set.

The `amp=False` parameter disables automatic mixed precision training, which is more about training efficiency and has less impact on the generation of validation graphs. It's unlikely to be the cause of your issue.

Regarding the validation code, when I mentioned reviewing it, I was referring to ensuring that the validation dataset is being loaded and processed correctly during training. However, since you're using the `.val()` function provided by our API, direct modification isn't necessary.

As for hyperparameters affecting loss graphs, they typically influence the training process rather than the logging of validation results. However, incorrect hyperparameter settings can lead to poor model convergence, which might indirectly affect the graphs.

To address your concerns:
- Ensure your validation set is distinct from your training set.
- Verify that your dataset is correctly labeled and formatted.
- Check that your `dataset.yaml` paths are accurate and that the file structure matches the expected format.

If the problem persists after these checks, please provide further details about your dataset and training configuration, and we can continue to troubleshoot together. Your collaboration is key to improving YOLOv8! 🌟

**Comment by Jes46** [comment]
Hi! Sorry for the delay, I was busy.
I just trained again my model, but this time with different images to train and val and I still have the same issue as I said before. 
Maybe, the following images could help to find the problem.

In these images you can see how my model doesn't perform validations during the training **_model.train(.......)_** and this happens with all val_batch_labels/pred collages.

**val_batch2_labels**
![val_batch2_labels](https://github.com/ultralytics/ultralytics/assets/150959706/5aa00430-5b05-49c5-8071-5a1ada9c292a)
**val_batch2_pred**
![val_batch2_pred](https://github.com/ultralytics/ultralytics/assets/150959706/357acfca-a693-477f-94d6-669ff5ea5c69)

BUT when I call **_model.val(.........)_** everything changes, and my model makes validations as you can see in the following images.

**val_batch2_labels**
![val_batch2_labels](https://github.com/ultralytics/ultralytics/assets/150959706/69d3a0b8-05c3-460a-bcc1-ec6a7ede76d3)
**val_batch2_pred**
![val_batch2_pred](https://github.com/ultralytics/ultralytics/assets/150959706/f03fe25a-3a44-479c-9589-eab8a234b1c8)

What should I do, would it be a solution to uninstall Ultralytics and install again,  or should I change any file associated with train mode?

Thanks again and have a nice day.

**Comment by glenn-jocher** [comment]
@Jes46, thanks for the update and for trying out a different set of images for training and validation. The fact that your model performs validations correctly when using `model.val()` but not during `model.train()` is indeed peculiar.

It's unlikely that reinstalling Ultralytics will resolve this issue, as it seems more related to the training configuration or a potential bug in the training loop.

Here's what I suggest:
- Double-check your training configuration to ensure all parameters are set correctly.
- Make sure your dataset is properly prepared and that the validation set is not empty or incorrectly labeled.
- Look into the training logs for any warnings or errors that might indicate what's going wrong during the validation steps within the training loop.

If the issue remains unresolved, please open an issue on the Ultralytics YOLOv8 GitHub repository with the details of your training configuration and dataset structure. This will allow the community to investigate further and provide a more targeted solution.

Your efforts are valuable to us, and we're committed to helping you resolve this. Have a great day! 🛠️

**Comment by Jes46** [comment]
Thanks again, I gonna do that. It will be better.

**Comment by shz998** [comment]
> Thanks again, I gonna do that. It will be better.

You are talking to a bot.

**Comment by glenn-jocher** [comment]
@Jes46 you're welcome! If you have any further questions or need assistance after reviewing your setup, feel free to reach out. Good luck with your training! 🚀

**Comment by mrsiro789** [comment]
@Jes46, hello, i have same situation, how do you fix it. Help me please

**Comment by glenn-jocher** [comment]
@Jes46 hello! Thanks for reaching out. 😊 It sounds like adjusting the batch size could indeed help with your situation. Increasing the batch size allows the model to have more data points for calculating the loss more reliably. Here's a quick example on how you might adjust it in your training command:

```bash
yolo train data=your_dataset.yaml model=yolov8n.pt batch=16
```

Just ensure that your GPU has enough memory to handle the larger batch size. Good luck with your training! 🚀
